tests/integration/cmd_push_test.go

```diff
@@ -0,0 +1,209 @@
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+			output := helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
+
+			// rename a file and push
+			helper.RenameFile(filepath.Join(context, "/nodejs-ex", "README.md"), filepath.Join(context, "/nodejs-ex", "NEW-README.md"))
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			// get the name of running pod
+			podName := oc.GetRunningPodNameOfComp(cmpName, project)
+
+			// verify that the new file was pushed
+			stdOut := oc.ExecListDir(podName, project)
+
+			// TODO enable once https://github.com/openshift/odo/issues/1354 is resolve
+			//Expect(stdOut).To(Not(ContainSubstring("README.md")))
+
+			Expect(stdOut).To(ContainSubstring("NEW-README.md"))
+
+			// rename a folder and push
+			helper.RenameFile(filepath.Join(context, "/nodejs-ex", "/tests"), filepath.Join(context, "/nodejs-ex", "/testing"))
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			// verify that the new file was pushed
+			stdOut = oc.ExecListDir(podName, project)
+
+			// TODO enable once https://github.com/openshift/odo/issues/1354 is resolve
+			//Expect(stdOut).To(Not(ContainSubstring("tests")))
+
+			Expect(stdOut).To(ContainSubstring("testing"))
+		})
+
+		It("should not build when changes are detected in a ignored file", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+
+			// create the .odoignore file and push
+			ignoreFilePath := filepath.Join(context, "nodejs-ex", ".odoignore")
+			if err := helper.CreateFileWithContent(ignoreFilePath, ".git\n*.md"); err != nil {
+				fmt.Printf("the .odoignore file was not created, reason %v", err.Error())
+			}
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			// modify a ignored file and push
+			helper.ReplaceString(filepath.Join(context+"/nodejs-ex"+"/README.md"), "This example will serve a welcome page", "This is a example welcome page!")
+			output := helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+			Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
+
+			// test ignores using the flag
+			output = helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex", "--ignore", "*.md")
+			Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
+		})
+
+		It("should build when no changes are detected in the directory and force flag is enabled", func() {
```

Comment:
Seems spec description does not follow the steps. 
```
Expect(output).To(Not(ContainSubstring("No file changes detected, skipping build")))
```
What i understand form the assertion is that even ```-f``` won't work to rebuild if there is no change. @mik-dass Can you please clarify